src/commands/createWebApp/setPostPromptDefaults.ts

```diff
@@ -37,49 +37,60 @@ export async function setPostPromptDefaults(wizardContext: IWebAppWizardContext,
+                    return plan.name && plan.name.includes(defaultPlanName) && getResourceGroupFromId(nonNullProp(plan, 'id')).includes(defaultGroupName);
+                });
+
+                // when using appServicePlans.list, the numOfSites are all set to 0 so individually get each plan and look for one with less than 3 sites
+                for (const plan of defaultPlans) {
+                    if (plan.name) {
+                        const groupName: string = getResourceGroupFromId(nonNullProp(plan, 'id'));
+                        const fullPlanData: WebSiteManagementModels.AppServicePlan | undefined = await tryGetAppServicePlan(client, groupName, plan.name);
+                        if (fullPlanData && matchesTier(fullPlanData, newSkuTier) && !checkPlanForPerformanceDrop(fullPlanData)) {
+                            wizardContext.newResourceGroupName = groupName;
+                            wizardContext.newPlanName = plan.name;
+                            break;
+                        }
                     }
                 }
-            }
 
-            wizardContext.newResourceGroupName = wizardContext.newResourceGroupName || await siteNameStep.getRelatedName(wizardContext, defaultGroupName);
-            if (!wizardContext.newResourceGroupName) {
-                throw new Error(localize('noUniqueNameRg', 'Failed to generate unique name for resources. Use advanced creation to manually enter resource names.'));
-            }
+                wizardContext.newResourceGroupName = wizardContext.newResourceGroupName || await siteNameStep.getRelatedName(wizardContext, defaultGroupName);
+                if (!wizardContext.newResourceGroupName) {
+                    throw new Error(localize('noUniqueNameRg', 'Failed to generate unique name for resources. Use advanced creation to manually enter resource names.'));
+                }
 
-            wizardContext.newPlanName = await siteNameStep.getRelatedName(wizardContext, defaultPlanName);
-            if (!wizardContext.newPlanName) {
-                throw new Error(localize('noUniqnueNameAsp', 'Failed to generate unique name for app service plan. Use advanced creation to manually enter plan names.'));
+                wizardContext.newPlanName = await siteNameStep.getRelatedName(wizardContext, defaultPlanName);
+                if (!wizardContext.newPlanName) {
+                    throw new Error(localize('noUniqnueNameAsp', 'Failed to generate unique name for app service plan. Use advanced creation to manually enter plan names.'));
+                }
             }
+        } else {
+            wizardContext.newResourceGroupName = defaultGroupName;
+            wizardContext.newPlanName = defaultPlanName;
+        }
+    } catch (error) {
+        if (parseError(error).errorType === 'AuthorizationFailed') {
+            // setting these values to some arbitrary values as ResourceGroupCreateStep & AppServicePlanCreateStep expects these to be not null
```

Comment:
Here's what I would say:
```suggestion
            // User has restricted permissions for something. We'll use the default values for now and let the  `ResourceGroupCreateStep` & `AppServicePlanCreateStep` handle the 'AuthorizationFailed' error down the line
```